File: tests/rustdoc-js/search-method-disambiguate.js
// exact-check
// ignore-order
// ignore-tidy-linelength

const FILTER_CRATE = "search_method_disambiguate";

const EXPECTED = [
    {
        'query': 'MyTy -> bool',
        'others': [
            {
                'path': 'search_method_disambiguate::MyTy',
                'name': 'my_method',
                'href': '../search_method_disambiguate/struct.MyTy.html#impl-X-for-MyTy%3Cbool%3E/method.my_method'
            },
        ],
    },
    {
        'query': 'MyTy -> u8',
        'others': [
            {
                'path': 'search_method_disambiguate::MyTy',
                'name': 'my_method',
                'href': '../search_method_disambiguate/struct.MyTy.html#impl-X-for-MyTy%3Cu8%3E/method.my_method'
            },
        ],
    }
];

File: tests/rustdoc-js/search-method-disambiguate.rs
pub trait X {
    type InnerType;
    fn my_method(&self) -> Self::InnerType;
}

pub struct MyTy<T> {
    pub t: T,
}

impl X for MyTy<bool> {
    type InnerType = bool;
    fn my_method(&self) -> bool {
        self.t
    }
}

impl X for MyTy<u8> {
    type InnerType = u8;
    fn my_method(&self) -> u8 {
        self.t
    }
}
